crates/nu-protocol/src/parse_error.rs

```diff
@@ -349,6 +349,10 @@ pub enum ParseError {
         #[label = "parameter {0} needs to be '{1}' instead of '{2}'"] Span,
     ),
 
+    #[error("Default values should be constant expressions.")]
+    #[diagnostic(code(nu::parser::non_constant_default_value))]
+    NonConstantDefaultValue(#[label = "expected a constant value"] Span),
```

Comment:
I'd prefer not intorducing a new variant to `ParseError` because it's really large..

Reply:
@WindSoilder - I think for this particular case it's okay.

We can refactor how constants are handled in the future, and then we can try to simplify how constants error in a future PR.

Reply:
I agree. I considered using one of the existing errors instead, but I think that none of them is appropriate for this use case and unhelpful error messages are tragic.